sessionctx/variable/sysvar.go

```diff
@@ -660,7 +661,21 @@ var defaultSysVars = []*SysVar{
 		return nil
 	}},
 	{Scope: ScopeNone, Name: Hostname, Value: DefHostname},
-	{Scope: ScopeSession, Name: Timestamp, Value: "", skipInit: true},
+	{Scope: ScopeSession, Name: Timestamp, Value: "", skipInit: true, MinValue: 1, MaxValue: 2147483647, GetSession: func(s *SessionVars) (string, error) {
+		defaultTimestamp := GetSysVar(Timestamp).Value
+		if timestamp, ok := s.systems[Timestamp]; ok && timestamp != defaultTimestamp {
+			return timestamp, nil
+		}
+		timestamp := s.StmtCtx.GetOrStoreStmtCache(stmtctx.StmtNowTsCacheKey, time.Now()).(time.Time)
+		return types.ToString(float64(timestamp.UnixNano()) / float64(time.Second))
```

Comment:
I think this will be easier to read if you set the `Value: ""`  to `Value: "0"`, and then if the value != 0 you can generate a value and cache it?

Reply:
> I think this will be easier to read if you set the `Value: ""` to `Value: "0"`, and then if the value != 0 you can generate a value and cache it?

I am a little confused here, if the value in `s.systems[Timestamp]` is equal to `"0"` which is the default value for the timestamp sysvar, we should do `GetOrStoreStmtCache`. But if the value in `s.systems[Timestamp]` is not equal to `"0"`, we should return that value directly

Reply:
Your code looks good now. The Value is now "0".